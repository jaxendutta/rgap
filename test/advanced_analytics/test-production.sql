-- We add a WHERE clause to help with performance with a production dataset
SELECT 
  YEAR(rg.agreement_start_date) AS year,
  o.org AS funding_agency,
  COUNT(*) AS total_grants,
  SUM(rg.agreement_value) AS total_funding,
  AVG(rg.agreement_value) AS avg_funding
FROM ResearchGrant rg
JOIN Organization o ON rg.org = o.org
WHERE YEAR(rg.agreement_start_date) BETWEEN YEAR(CURDATE()) - 10 AND YEAR(CURDATE())
GROUP BY YEAR(rg.agreement_start_date), o.org
ORDER BY year, total_funding DESC;
