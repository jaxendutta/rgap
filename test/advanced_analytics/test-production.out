+------+----------------+--------------+---------------+---------------+
| year | funding_agency | total_grants | total_funding | avg_funding   |
+------+----------------+--------------+---------------+---------------+
| 2015 | CIHR           |         1589 | 1237963345.00 | 779083.288232 |
| 2015 | NSERC          |            1 |      63000.00 |  63000.000000 |
| 2016 | CIHR           |         1671 | 1173216208.00 | 702104.253740 |
| 2016 | SSHRC          |            1 |     112500.00 | 112500.000000 |
| 2017 | CIHR           |         1732 | 1016957222.00 | 587157.749423 |
| 2017 | SSHRC          |         3329 |  794419407.65 | 238636.049159 |
| 2017 | NSERC          |        10652 |  791677701.14 |  74321.977201 |
| 2018 | SSHRC          |         5982 | 1223739196.43 | 204570.243469 |
| 2018 | NSERC          |        13284 | 1065149917.18 |  80182.920595 |
| 2018 | CIHR           |         3189 | 1063322501.00 | 333434.462527 |
| 2019 | NSERC          |        14514 | 1521371381.65 | 104820.957810 |
| 2019 | SSHRC          |         5781 | 1108923695.96 | 191822.123501 |
| 2019 | CIHR           |         3330 | 1062117145.00 | 318954.097598 |
| 2020 | SSHRC          |         9721 | 1360344070.55 | 139938.696693 |
| 2020 | CIHR           |         3140 | 1004932702.00 | 320042.261783 |
| 2020 | NSERC          |        17423 |  968902993.95 |  55610.571885 |
| 2021 | SSHRC          |         5687 | 1227278377.18 | 215804.180971 |
| 2021 | CIHR           |         2816 | 1047613658.00 | 372021.895597 |
| 2021 | NSERC          |         7766 |  447915971.42 |  57676.535079 |
| 2022 | NSERC          |         8545 | 1500806393.75 | 175635.622440 |
| 2022 | CIHR           |         3454 | 1340519535.00 | 388106.408512 |
| 2022 | SSHRC          |         5874 | 1108986162.10 | 188795.737504 |
| 2023 | SSHRC          |         5697 | 2964737272.66 | 520403.242524 |
| 2023 | NSERC          |         9768 | 1222321155.71 | 125135.253451 |
| 2023 | CIHR           |         3945 | 1040531219.00 | 263759.497845 |
| 2024 | CIHR           |         3718 | 1162630622.00 | 312703.233459 |
| 2024 | NSERC          |         9114 | 1138775547.71 | 124947.942474 |
| 2024 | SSHRC          |         5062 |  985387462.45 | 194663.663068 |
| 2025 | SSHRC          |           19 |    1556000.00 |  81894.736842 |
| 2025 | NSERC          |           45 |    1035500.00 |  23011.111111 |
+------+----------------+--------------+---------------+---------------+
30 rows in set (0.57 sec)
